packages/node/src/integrations/anr/index.ts

```diff
@@ -0,0 +1,172 @@
+import { URL } from 'url';
+import { getCurrentHub } from '@sentry/core';
+import type { Contexts, Event, EventHint, Integration } from '@sentry/types';
+import { dynamicRequire, logger } from '@sentry/utils';
+import type { Worker, WorkerOptions } from 'worker_threads';
+import type { NodeClient } from '../../client';
+import { NODE_VERSION } from '../../nodeVersion';
+import type { Options, WorkerStartData } from './common';
+import { base64WorkerScript } from './worker-script';
+
+const DEFAULT_INTERVAL = 50;
+const DEFAULT_HANG_THRESHOLD = 5000;
+
+type WorkerNodeV14 = Worker & { new (filename: string | URL, options?: WorkerOptions): Worker };
+
+type WorkerThreads = {
+  Worker: WorkerNodeV14;
+};
+
+function log(message: string, ...args: unknown[]): void {
+  logger.log(`[ANR] ${message}`, ...args);
+}
+
+/**
+ * We need to use dynamicRequire because worker_threads is not available in node < v12 and webpack error will when
+ * targeting those versions
+ */
+function getWorkerThreads(): WorkerThreads {
+  return dynamicRequire(module, 'worker_threads');
+}
+
+/**
+ * Gets contexts by calling all event processors. This relies on being called after all integrations are setup
+ */
+async function getContexts(client: NodeClient): Promise<Contexts> {
+  let event: Event | null = { message: 'ANR' };
+  const eventHint: EventHint = {};
+
+  for (const processor of client.getEventProcessors()) {
+    if (event === null) break;
+    event = await processor(event, eventHint);
+  }
+
+  return event?.contexts || {};
+}
+
+interface InspectorApi {
+  open: (port: number) => void;
+  url: () => string | undefined;
+}
+
+/**
+ * Starts the node debugger and returns the inspector url.
+ *
+ * When inspector.url() returns undefined, it means the port is already in use so we try the next port.
+ */
+function startInspector(startPort: number = 9229): string | undefined {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const inspector: InspectorApi = require('inspector');
+  let inspectorUrl: string | undefined = undefined;
+  let port = startPort;
```

Comment:
Oh nice. Does it start at a reasonably high number? I wouldn't want it to use anything under 1024

Reply:
According to the documentation, it would search >= 1024 and <= 65535 port.

> . If you do this the kernel will select an unused port from the [ephemeral port range](https://en.wikipedia.org/wiki/Ephemeral_port).